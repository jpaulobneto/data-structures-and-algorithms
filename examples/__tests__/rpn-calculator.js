const RPNCalculator = require('../rpn-calculator');

describe('RPNCalculator', () => {
  it('is defined', () => {
    expect(RPNCalculator).toBeDefined();
  });

  it('can be initialized', () => {
    expect(() => new RPNCalculator()).not.toThrow();
  });

  it('loadExpression(expression)', () => {
    const rpnc = new RPNCalculator();

    rpnc.loadExpression('1,2,+,3,+');

    expect(rpnc.operands).toMatchSnapshot();
    expect(rpnc.operators).toMatchSnapshot();
  });

  it('calculate(expression)', () => {
    const rpnc = new RPNCalculator();
    const expected = [
      {
        expression: '5,8,+',
        result: 13,
      },
      {
        expression: '9,4,+,6,+',
        result: 19,
      },
      {
        expression: '9,4,+,6,+,2,+,5,+',
        result: 26,
      },
    ];

    expected.forEach(test => expect(rpnc.calculate(test.expression)).toEqual(test.result))
  });
});
