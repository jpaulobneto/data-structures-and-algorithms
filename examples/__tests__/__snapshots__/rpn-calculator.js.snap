// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RPNCalculator loadExpression(expression) 1`] = `
Stack {
  "size": 3,
  "top": Cell {
    "next": Cell {
      "next": Cell {
        "next": Cell {
          "next": null,
          "value": 1,
        },
        "value": 2,
      },
      "value": 3,
    },
    "value": null,
  },
}
`;

exports[`RPNCalculator loadExpression(expression) 2`] = `
Stack {
  "size": 2,
  "top": Cell {
    "next": Cell {
      "next": Cell {
        "next": null,
        "value": "+",
      },
      "value": "+",
    },
    "value": null,
  },
}
`;
